File: java_app/src/main/java/com/petnagy/java/example/injection/ApplicationModule.java
package com.petnagy.java.example.injection;

import com.petnagy.java.example.redux.AppReducer;
import com.petnagy.java.example.redux.AppState;
import com.petnagy.java.example.redux.LogMiddleware;
import com.petnagy.koredux.Store;

import java.util.Collections;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ApplicationModule {

    @Provides
    @Singleton
    public Store<AppState> provideStore() {
        return new Store<>(new AppReducer(), Collections.singletonList(new LogMiddleware()), new AppState(0));
    }

}

File: app/src/main/java/com/petnagy/app/injection/ApplicationModule.kt
package com.petnagy.app.injection

import com.petnagy.app.redux.AppReducer
import com.petnagy.app.redux.AppState
import com.petnagy.app.redux.LogMiddleware
import com.petnagy.koredux.Store
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class  ApplicationModule {

    @Provides
    @Singleton
    fun provideStore(): Store<AppState> = Store(reducer = AppReducer(), middlewareList = listOf(LogMiddleware()),
            initState = AppState(0))

}